File: library/src/main/java/zhan/library/widget/HeaderRelativeLayout.java
package zhan.library.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.RelativeLayout;
import zhan.library.delegate.ScrollHeaderDelegate;


public class HeaderRelativeLayout extends RelativeLayout {

    private ScrollHeaderDelegate mScrollHeaderDelegate;

    public HeaderRelativeLayout(Context context) {
        super(context);
        init();
    }

    public HeaderRelativeLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public HeaderRelativeLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        mScrollHeaderDelegate = new ScrollHeaderDelegate(this);
    }

    @Override
    public void computeScroll() {
        mScrollHeaderDelegate.computeScroll();
    }


    public void setScrollShow(boolean isScrollShow) {
        mScrollHeaderDelegate.setScrollShow(isScrollShow);
    }

    public boolean isScrollShow() {
        return mScrollHeaderDelegate.isScrollShow();
    }

    public void scrollShow() {
        mScrollHeaderDelegate.scrollShow();
    }

    public void setDuration(int duration) {
        mScrollHeaderDelegate.setDuration(duration);
    }
}

File: double-pull-delegate/src/main/java/zhan/pull/widget/HeaderRelativeLayout.kt
package zhan.pull.widget

import android.content.Context
import android.util.AttributeSet
import android.widget.RelativeLayout
import zhan.pull.delegate.ScrollHeaderDelegate


class HeaderRelativeLayout : RelativeLayout {

    private var mScrollHeaderDelegate: ScrollHeaderDelegate? = null

    var isScrollShow: Boolean
        get() = mScrollHeaderDelegate!!.isScrollShow
        set(isScrollShow) {
            mScrollHeaderDelegate!!.isScrollShow = isScrollShow
        }

    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init()
    }

    private fun init() {
        mScrollHeaderDelegate = ScrollHeaderDelegate(this)
    }

    override fun computeScroll() {
        mScrollHeaderDelegate!!.computeScroll()
    }

    fun scrollShow() {
        mScrollHeaderDelegate!!.scrollShow()
    }

    fun setDuration(duration: Int) {
        mScrollHeaderDelegate!!.setDuration(duration)
    }
}
